docs/user-guide/creating-application/workflow/ci-pipeline.md

```diff
@@ -1,53 +1,120 @@
+
+![](../../../.gitbook/assets/ci-pipeline-1.png)
+
 ## A. Continuous Integration
 
-CI related configuration is provided under this section. Key concepts of CI pipeline are: 
+Click on Continuous Integration, a prompt comes up in which we need to provide our custom configurations. Below is the description of some configurations which are required.
 
 | Key | Description |
 | :--- | :--- |
 | `Pipeline Name` | Name of the pipeline |
+| `Pipeline Execution (Advanced)` | Select from automatic or manual execution depending upon your use-case |
 | `Source Type` | Select the source through which the CI Pipeline will be triggered  |
-| `Branch Name/Tag Regex` | Enter branch name/tag regex. |
-| `Stages` | 1.**Pre-build Stages**- Scripts to be executed before building an image.   2.**Docker build Stages**- Provide a new argument and override an old argument in key-value pair.    3. **Post-build Stages**- Scripts to be executed after building image |
-| `Scan for vulnerabilities` | It will scan your image and find if any vulnerabilities present |
+| `Stages (Advanced)` | 1.**Pre-build Stages**- Scripts to be executed before building an image.   2.**Docker build Stages**- Provide a new argument and override an old argument in key-value pair.     3. **Post-build Stages**- Scripts to be executed after building image |
+| `Scan for vulnerabilities (Advanced)` | It will scan your image and find if any vulnerabilities present |
+
+
+[Note] Options such as pipeline execution, stages and scan for vulnerabilities, will be visible after clicking on advanced options present in the bottom left corner.
+
+### I. Pipeline Name
+Pipeline name is an auto-generated name which can also be renamed by clicking on Advanced options.
+
+### II. Pipeline Execution
+You can select the method you want to execute the pipeline. By default the value is automatic. In this case it will get automatically triggered if any changes are made to the respective git repository. You can set it to manual if you want to trigger the pipeline manually.
+
+### III. Source Type
+In source type, we can observe that we have three types of mechanisms which can be used for building your CI Pipeline. In the drop-down you can observe we have Branch Fixed, Pull Request and Tag Creation. 
+
+![](../../../.gitbook/assets/ci-pipeline-2.png)
+
+#### i) Branch Fixed
+If you select the Branch Fixed as your source type for building CI Pipeline, then you need to provide the corresponding Branch Name.
+
+Branch Name is the name of the corresponding branch (eg. main or master, or any other branch)
+
+#### ii) Pull Request
+[Note] It only works if Git Host is Github or Bitbucket Cloud as of now. In case you need support for any other Git Host, please create a [github issue](https://github.com/devtron-labs/devtron/issues).
```

Comment:
I hope you have added git host documentation as well.

Reply:
Git Host doc will be added in Global Configuration part. Will do it.